### ISSUE TITLE
Integrate upstream change 

### ISSUE THREAD

**Issue by jcfr** [commontk/PythonQt#3]
The text reported below has been sent by Florian Link from Mesvis lab after he integrated the CTK contributions:

```
- Renamed set/enableRedirectStdInCallBack to set/enableRedirectStdInCallback 
(since other methods use Callback spelling)
See https://github.com/commontk/CTK/issues/156
EDIT: DONE

- Removed VTK wrapping from PythonQt
EDIT: DONE

- Removed VTK wrapping from PythonQtConversion
EDIT: DONE

- Removed VTK from CMake filelist
EDIT: DONE

- Added new PythonQt::addWrapperFactory(PythonQtForeignWrapperFactory*) 
which allows to implement the VTK integration
  (or any other wrapper integration) externally (so you can do it in CTK)
See https://github.com/commontk/CTK/issues/157
EDIT: DONE

- PyMapping_Items -> PyObject_CallMethod fix not taken (PythonQtConversion.cpp), 
since it is not a good idea to expand PyMapping_Items just to get rid of the const char* 
warning (PyMapping_Items might change anytime in the future)

- renamed dPython.h to PythonQtPythonInclude and made it optional by adding a define check:  
  PYTHONQT_USE_RELEASE_PYTHON_FALLBACK and added that define to the CMake file list. 
See https://github.com/commontk/PythonQt/issues/4

- Changed polymorphic handler signature to const char** so that no const_cast<char*>
 is needed anymore
```

**Comment by jeinarsson** [comment]
Thanks for working on all this.

I compiled Python 2.7.2 + Qt 4.8 PythonQt from this repo with msvc2008. To get Debug mode working (that is, not mixing runtimes and crashing), I had to remove the hack in dPython.h completely, and link to python27_d.lib manually. I guess this is what PYTHONQT_USE_RELEASE_PYTHON_FALLBACK will fix in the future.

Moreover the patch in issue #2 is needed for compilation.

What is the recommended way to compile PythonQt_QtAll now?  Could it be merged into PythonQt? 
EDIT: PythonQt_Wrap_QtAll added in 97df3b08

I don't understand all of the dependencies.. I got Release mode working via the QtCreator project, but the Debug project keeps complaining about the python library thus far..
EDIT: Using Qt Creator for the extension actually worked out now.. However, naming the debug lib PythonQt_d.lib and putting it all into CMake would be nice. If I manage I'll ping you again...

EDIT: and is there a discussion channel for this project? I know I'm drifting off topic in here..
